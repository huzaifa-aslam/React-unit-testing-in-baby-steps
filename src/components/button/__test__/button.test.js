import React from "react";
import ReactDOM from "react-dom";
import Button from "../button";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
afterEach(cleanup);
it("render without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button />, div);
});

it("render button correctly", () => {
  const { getByTestId } = render(<Button label="click me" />);
  expect(getByTestId("test")).toHaveTextContent("click me");
});

it("render button correctly", () => {
  const { getByTestId } = render(<Button label="save" />);
  expect(getByTestId("test")).toHaveTextContent("save");
});

it("matches snapshot 1", () => {
  const tree = renderer.create(<Button label="save" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot 2", () => {
  const tree = renderer.create(<Button label="click me please" />).toJSON();
  expect(tree).toMatchSnapshot();
});
