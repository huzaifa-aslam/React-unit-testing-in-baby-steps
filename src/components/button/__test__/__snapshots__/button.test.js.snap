// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <button
    data-testid="test"
  >
  Hi
    save
  </button>
</div>
`;

exports[`matches snapshot 1 1`] = `
<div>
  <button
    data-testid="test"
  >
    Hi 
    save
  </button>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div>
  <button
    data-testid="test"
  >
    Hi 
    click me please
  </button>
</div>
`;
